### ISSUE TITLE
[MPMD-257] Allow to disable analysisCache completely, avoid warnings

### ISSUE THREAD

**Issue by jira-importer** [apache/maven-pmd-plugin#594]
**[Roman Ivanov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=romani)** opened **[MPMD-257](https://issues.apache.org/jira/browse/MPMD-257?redirect=false)** and commented

in plugn where is property
https://maven.apache.org/plugins/maven-pmd-plugin/pmd-mojo.html#analysisCache
that suppose to enable/disable cache

but it is used only to allow set for cache location:
https://github.com/apache/maven-pmd-plugin/blob/89687854bc1c965f26adbc78410f75e631f66088/src/main/java/org/apache/maven/plugins/pmd/PmdReport.java#L678

Pmd allow to skip warning is "-nocahe" is defined.
https://github.com/pmd/pmd/blob/451815bf0a790805a6a38c1c8bfe798f2a5f04bd/pmd-core/src/main/java/net/sourceforge/pmd/PMD.java#L300

Detected:

```java
$ mvn pmd:pmd -Dpmd.analysisCache=false
...
[WARNING] This analysis could be faster, please consider using Incremental Analysis: https://pmd.github.io/pmd-6.1.0/pmd_userdocs_getting_started.html#incremental-analysis
..
```

Expected: no WARNING in output

 no WARNING 'This analysis could be faster ...' when **analysisCache=false**

 

---

**Issue Links:**
- [MPMD-261](https://issues.apache.org/jira/browse/MPMD-261) Upgrade to PMD 6.4.0
 (_**"is blocked by"**_)

1 votes, 4 watchers

**Comment by jira-importer** [comment]
**[Andreas Dangel](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=adangel)** commented

This requires PMD 6.2.0:

https://pmd.github.io/pmd-6.2.0/pmd_release_notes.html#disable-incremental-analysis
